### PR TITLE
Remove toolkit functions deprecated in 3.1

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

This pulls out parts of #15890 to make that PR more managable and easier to debug the test failures.

### CODE REVIEW

File: doc/api/prev_api_changes/api_changes_3.1.0.rst [hunk 1]
```diff
@@ -863,7 +863,7 @@ Use the standard library's docstring manipulation tools instead, such as
 These are considered internal and will be removed from the public API in a
 future version.
 
-- `.projections.process_projection_requirements`
+- ``projections.process_projection_requirements``
```
Comment (anntzer) [inline doc/api/prev_api_changes/api_changes_3.1.0.rst]: these should technically be in a separate PR but I'm not going to be nitpicky with that.

Comment (timhoffm) [reply doc/api/prev_api_changes/api_changes_3.1.0.rst]: If one wants to be very strict yes, but it's just changing link to literal, which can be done anytime is not strictly related to the deprecation of that function. I saved the hassle of cherry-pickling parts of the file. :smile:
